provider/resource_service.go

```diff
@@ -434,57 +722,26 @@ func service(ctx context.Context, d *schema.ResourceData, clientConfig models.Cl
 		}
 	}
 
-	//compute kpiIds for dataResource
-	if d.HasChange("kpi") {
-		kpisOld, kpisNew := d.GetChange("kpi")
-		kpiOldKeys := map[string]string{}
-
-		for _, kpi := range kpisOld.(*schema.Set).List() {
-			kpiData := kpi.(map[string]interface{})
-
-			// kpiid is important to save for historical raw data. Historical raw data makes sense,
-			// until base search & metris is same
-			internalIdentifier, err := getKpiHashKey(kpiData)
-			if err != nil {
-				return nil, err
-			}
-
-			kpiOldKeys[internalIdentifier] = kpiData["id"].(string)
-		}
-		for _, kpi := range kpisNew.(*schema.Set).List() {
-			kpiData := kpi.(map[string]interface{})
-
-			internalIdentifier, err := getKpiHashKey(kpiData)
-			if err != nil {
-				return nil, err
-			}
-
-			if existingKpiId, ok := kpiOldKeys[internalIdentifier]; ok {
-				kpiData["id"] = existingKpiId
-			} else {
-				kpiData["id"], _ = GenerateUUID(internalIdentifier)
-			}
-		}
-		err := d.Set("kpi", kpisNew)
-		if err != nil {
-			return nil, err
+	itsiKpis := []map[string]interface{}{}
+	for _, kpi := range m.KPIs {
+		if kpi.ID.IsUnknown() {
+			uuid, _ := uuid.GenerateUUID()
+			kpi.ID = types.StringValue(uuid)
```

Comment:
we can't use getKpiHashKey to get the KPI ID anymore?

Reply:
We can, we just didn't do this in prev version:
```
			if existingKpiId, ok := kpiOldKeys[internalIdentifier]; ok {
				kpiData["id"] = existingKpiId
			} else {
				kpiData["id"], _ = GenerateUUID(internalIdentifier)
			}
```

Reply:
```
// testOverridable
var GenerateUUID = func(internalIdentifier string) (string, error) {
	return uuid.GenerateUUID()
}
```

Reply:
getKpiHashKey was used for internal mapping (it is still there in the plan modifier https://github.com/TiVo/terraform-provider-splunk-itsi/pull/418/files#diff-6bcac826cdcc1c2d664f422c46abbafc7b31fd82f90dfdb8ca063134fa250af2R343)

Reply:
AFAIK it was done that way, in order to follow "We don't know how it is saved and used in Mongo, if UI generates UUID, lets follow that strategy"

Reply:
got it. modifyplan does this instead, I forgot how the new plugin does this.